### PR TITLE
Fixing Windows10 build for V2 client library

### PR DESCRIPTION
**Pull request by tanmayv25**

### CODE REVIEW

File: src/clients/c++/experimental_api_v2/library/common.cc [hunk 1]
```diff
@@ -228,7 +228,7 @@ InferInput::GetNext(
 
   while ((bufs_idx_ < bufs_.size()) && (size > 0)) {
     const size_t buf_byte_size = buf_byte_sizes_[bufs_idx_];
-    const size_t csz = std::min(buf_byte_size - buf_pos_, size);
+    const size_t csz = (std::min)(buf_byte_size - buf_pos_, size);
```
Comment (deadeyegoodwin) [inline src/clients/c++/experimental_api_v2/library/common.cc:231]: Strange thing for windows to require... seems like a bug in their compiler

File: src/clients/c++/experimental_api_v2/library/common.h [hunk 2]
```diff
@@ -40,6 +40,9 @@
 #include <string>
 #include <thread>
 #include <vector>
+#ifdef _WIN32
+#include <algorithm>
```
Comment (deadeyegoodwin) [inline src/clients/c++/experimental_api_v2/library/common.h:44]: I this file requires something from <algorithm> we should just include it for all platforms

File: src/clients/c++/experimental_api_v2/library/common.cc (hunk 1 above)
Comment (tanmayv25) [reply src/clients/c++/experimental_api_v2/library/common.cc:231]: In one of the standard headers for windows, min and max are defined as MACROS that takes two arguments as well. This causes a conflict between the macro and std::min function. Enclosing the (std::min)/(std::max) ensures the function is selected and not the macro.

File: src/clients/c++/experimental_api_v2/library/common.h [hunk 3]
```diff
@@ -40,6 +40,7 @@
 #include <string>
 #include <thread>
 #include <vector>
+#include <algorithm>
```
Comment (deadeyegoodwin) [inline src/clients/c++/experimental_api_v2/library/common.h:43]: Probably need to run formatter. No need to restart CI after doing that.

Comment (tanmayv25) [reply src/clients/c++/experimental_api_v2/library/common.h]: Done

**Pull request merged**